test/e2e/affinity_group.go

```diff
@@ -73,6 +73,14 @@ func AffinityGroupSpec(ctx context.Context, inputGetter func() CommonSpecInput)
 		affinityIds = executeTest(ctx, input, namespace, specName, clusterResources, "anti")
 	})
 
+	It("Should have host affinity group when affinity is soft-pro", func() {
+		affinityIds = executeTest(ctx, input, namespace, specName, clusterResources, "soft-pro")
+	})
+
+	It("Should have host affinity group when affinity is soft-anti", func() {
+		affinityIds = executeTest(ctx, input, namespace, specName, clusterResources, "soft-anti")
+	})
+
```

Comment:
@FarnazBGH Can you add a check for running this only for supported cloudstack versions else skip?

Reply:
The cloudStack version Check for soft affinities is added. The result for cloudStack Versions lower than 4.18 will be:

```
S [SKIPPING] [0.492 seconds]
When testing affinity group
~/cluster-api-provider-cloudstack/test/e2e/affinity_group_test.go:27
  Should have host affinity group when affinity is soft-pro [It]
~/cluster-api-provider-cloudstack/test/e2e/affinity_group.go:76

  Soft affinity groups are only supported in CloudStack version 4.18 or higher.

~/cluster-api-provider-cloudstack/test/e2e/affinity_group.go:79
------------------------------
When testing affinity group
  Should have host affinity group when affinity is soft-anti
~/cluster-api-provider-cloudstack/test/e2e/affinity_group.go:84
STEP: Getting a CloudStack client secret
STEP: Creating a CloudStack client
STEP: Creating a namespace for hosting the "affinity-group" test spec
INFO: Creating namespace affinity-group-psmshm
INFO: Creating event watcher for namespace "affinity-group-psmshm"

S [SKIPPING] [0.417 seconds]
When testing affinity group
~/cluster-api-provider-cloudstack/test/e2e/affinity_group_test.go:27
  Should have host affinity group when affinity is soft-anti [It]
~//cluster-api-provider-cloudstack/test/e2e/affinity_group.go:84

  Soft affinity groups are only supported in CloudStack version 4.18 or higher.

~/cluster-api-provider-cloudstack/test/e2e/affinity_group.go:87
```